snafu/benchmarks/dnsperf/dnsperf.py

```diff
@@ -0,0 +1,224 @@
+        ),
+        ConfigArgument(
+            "-w",
+            "--time-window-size",
+            dest="time_window_size",
+            env_var="time_window_size",
+            help="Time window size for test",
+            default=".01",
+        ),
+        ConfigArgument("--timeout", help="Length of timeout in seconds", dest="timeout_len", default="5"),
+        ConfigArgument(
+            "-m",
+            "--transport-mode",
+            dest="transport_mode",
+            help="set transport mode: udp, tcp, dot",
+            default="udp",
+        ),
+        ConfigArgument("--network-policy", dest="networkpolicy", env_var="networkpolicy"),
+        ConfigArgument("--pod-id", dest="pod_id", default=None),
+        ConfigArgument("--node-id", dest="node_id", default=None),
+    )
+
+    def setup(self) -> bool:
+        """Setup dnsperf benchmark."""
+        self.logger.info("Setting up dnsperf benchmark.")
+        self.config.parse_args()
+        self.config.load_sequence.append(None)
+        # dynamically set timeout length for template parser
+        self.output_template = (
+            self.output_template + "\n"
+            "<vars>"
+            "default_values = {"
+            f"'rtt_s': {self.config.timeout_len}"
+            "}"
+            "</vars>\n"
+        )
+        return True
+
+    def cleanup(self) -> bool:
+        """Clean up artifacts from the dnsperf benchmark."""
+        self.logger.info("Cleaning up dnsperf benchmark.")
+        return True
+
+    def collect(self) -> Iterable[BenchmarkResult]:
+        """Run the dnsperf benchmark and collect results."""
+
+        self.logger.info("Starting dnsperf")
+        for load in self.config.load_sequence:
+            cmd = [
+                "dnsperf",
+                "-v",  # print latency information for each query
+                "-s",
+                self.config.address,
+                "-p",
+                self.config.port,
+                "-d",
+                self.config.query_filepath,
+                "-c",
+                self.config.clients,
+                "-l",
+                self.config.time_window_size,
+                "-t",
+                self.config.timeout_len,
+                "-m",
+                self.config.transport_mode,
+            ]
+
+            if load:
+                cmd = [*cmd, "-Q", str(load)]
+
+            sample: ProcessSample = sample_process(
+                cmd, self.logger,
+            )
+
+            if not sample.success:
+                self.logger.critical(f"dnsperf failed to complete! Got results: {sample}\n")
+                return None
+            elif sample.successful.stdout is None:
+                self.logger.critical(
+                    (
+                        "dnsperf ran successfully, but did not get output on stdout.\n"
+                        f"Got results: {sample}\n"
+                    )
+                )
+                return None
+
```

Comment:
Can we add a log message here saying that dnsperf did run successfully? Maybe at an info level?

Reply:
yep